src/Elasticsearch5/SearchAdapter/Aggregation/Interval.php

```diff
@@ -0,0 +1,238 @@
+{
+    /**
+     * Minimal possible value
+     */
+    const DELTA = 0.005;
+
+    /**
+     * @var ConnectionManager
+     */
+    private $connectionManager;
+
+    /**
+     * @var FieldMapperInterface
+     */
+    private $fieldMapper;
+
+    /**
+     * @var Config
+     */
+    private $clientConfig;
+
+    /**
+     * @var string
+     */
+    private $fieldName;
+
+    /**
+     * @var string
+     */
+    private $storeId;
+
+    /**
+     * @var array
+     */
+    private $entityIds;
+
+    /**
+     * @var SearchIndexNameResolver
+     */
+    private $searchIndexNameResolver;
+
+    /**
+     * @param ConnectionManager $connectionManager
+     * @param FieldMapperInterface $fieldMapper
+     * @param Config $clientConfig
+     * @param SearchIndexNameResolver $searchIndexNameResolver
+     * @param string $fieldName
+     * @param string $storeId
+     * @param array $entityIds
+     */
+    public function __construct(
+        ConnectionManager $connectionManager,
+        FieldMapperInterface $fieldMapper,
+        Config $clientConfig,
+        SearchIndexNameResolver $searchIndexNameResolver,
+        string $fieldName,
+        string $storeId,
+        array $entityIds
+    ) {
+        $this->connectionManager = $connectionManager;
+        $this->fieldMapper = $fieldMapper;
+        $this->clientConfig = $clientConfig;
+        $this->fieldName = $fieldName;
+        $this->storeId = $storeId;
+        $this->entityIds = $entityIds;
+        $this->searchIndexNameResolver = $searchIndexNameResolver;
+
+        parent::__construct(
+            $connectionManager,
+            $fieldMapper,
+            $clientConfig,
+            $searchIndexNameResolver,
+            $fieldName,
+            $storeId,
+            $entityIds,
+
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function load($limit, $offset = null, $lower = null, $upper = null)
+    {
+
+        $from = ['gte' => 0];
+        $to = ['lt' => 0];
+
+        if ($lower) {
+            $from = ['gte' => $lower - self::DELTA];
+        }
+
+        if ($upper) {
+            $to = ['lt' => $upper - self::DELTA];
+        }
+
+        $requestQuery = $this->prepareBaseRequestQuery($from, $to);
+        $requestQuery = array_merge_recursive(
+            $requestQuery,
+            ['body' => ['stored_fields' => [$this->fieldName], 'size' => $limit]]
+        );
+
+        if ($offset) {
+            $requestQuery['body']['from'] = $offset;
+        }
+
+        $queryResult = $this->connectionManager->getConnection()
+            ->query($requestQuery);
+
+        return $this->arrayValuesToFloat($queryResult['hits']['hits'], $this->fieldName);
+    }
+
+
+    /**
+     * @inheritdoc
+     */
+    public function loadPrevious($data, $index, $lower = null)
+    {
+
+        $from = ['gte' => 0];
+        $to = ['lt' => 0];
+
+        if ($lower) {
+            $from = ['gte' => $lower - self::DELTA];
+        }
+        if ($data) {
```

Comment:
Please separate blocks with the new line here and in other places in the code too.